File: lms_mobile_web/lib/features/admin/leave/domain/models/leave_status.dart
/// 휴가 승인 상태
enum LeaveStatus {
  /// 대기 중
  pending('PENDING', '대기중'),

  /// 승인됨
  approved('APPROVED', '승인'),

  /// 반려됨
  rejected('REJECTED', '반려'),

  /// 취소됨
  cancelled('CANCELLED', '취소');

  final String value;
  final String displayName;

  const LeaveStatus(this.value, this.displayName);

  static LeaveStatus fromString(String value) {
    return LeaveStatus.values.firstWhere(
      (status) => status.value == value,
      orElse: () => LeaveStatus.pending,
    );
  }

  @override
  String toString() => displayName;
}

File: domain/src/main/kotlin/com/lms/domain/model/leave/LeaveStatus.kt
package com.lms.domain.model.leave

/**
 * 휴가 신청 상태
 */
enum class LeaveStatus(val description: String) {
    PENDING("승인 대기"),
    APPROVED("승인됨"),
    REJECTED("거부됨"),
    CANCELLED("취소됨")
}
